p2p/src/net/mod.rs

```diff
@@ -50,10 +50,13 @@ where
 
     /// One one more peers have expired
     PeerExpired(Vec<AddrInfo<T>>),
+
+    /// Message received from a Gossipsub topic
+    MessageReceived(GossipsubTopic, message::Message),
 }
 
 #[derive(Debug)]
-pub enum GossipSubTopic {
+pub enum GossipsubTopic {
```

Comment:
Why is it necessary to have a "topic"? Don't we have a message type already?

Reply:
We need to be able to subscribe to a topic